### PR TITLE
Documentation: Add a note on construction of `Device` objects in Kernel

### PR DESCRIPTION
**Pull request by supercomputer7**

### CODE REVIEW

File: Documentation/Kernel/DevelopmentGuidelines.md [hunk 1]
```diff
@@ -198,6 +198,31 @@ We allocate them in `Kernel/API/MajorNumberAllocation.h`, based on this set of r
...
+        NUM_TRANSFER_REGION_PAGES * PAGE_SIZE,
+        "VIRGL3D kernel upload buffer"sv,
+        Memory::Region::Access::ReadWrite,
+        AllocationStrategy::AllocateNow));
+    auto kernel_context_id = TRY(adapter.create_context());
+    return TRY(DeviceManagement::try_create_device<VirtIOGPU3DDevice>(adapter, move(region_result), kernel_context_id));
+}
+```
+
+The reason for using `DeviceManagement` `try_create_device` method, is because that method
```
Comment (nico) [inline Documentation/Kernel/DevelopmentGuidelines.md:222]: nit: no comma on this line

File: Documentation/Kernel/DevelopmentGuidelines.md [hunk 2]
```diff
@@ -198,6 +198,31 @@ We allocate them in `Kernel/API/MajorNumberAllocation.h`, based on this set of r
 6. The allocation must be inserted by keeping the ascending order of the allocated major numbers
    in the appropriate enum and associated to-StringView function.
 
+## Construction of `Device`-based objects
+
+Currently, we have many devices that are either inserted in boot time but also devices that could be inserted
```
Comment (nico) [inline Documentation/Kernel/DevelopmentGuidelines.md:203]: nit: s/in boot time/at boot time/

**Review by nico — APPROVED** [review]
thanks!

**Comment by supercomputer7** [comment]
CI is flaky and failing of it has nothing to do with this change.

File: Documentation/Kernel/DevelopmentGuidelines.md [hunk 3]
```diff
@@ -198,6 +198,31 @@ We allocate them in `Kernel/API/MajorNumberAllocation.h`, based on this set of r
...
+
+Currently, we have many devices that are either inserted at boot time but also devices that could be inserted
+afterwards.
+
+To make it easier writing device drivers, when constructing an object from a `Device`-derived class, the usual pattern is to use `DeviceManagement` `try_create_device` method.
+For example, constructing a `VirtIOGPU3DDevice` is done this way:
+```c++
+ErrorOr<NonnullLockRefPtr<VirtIOGPU3DDevice>> VirtIOGPU3DDevice::try_create(VirtIOGraphicsAdapter& adapter)
+{
+    // Setup memory transfer region
```
Comment (nico) [inline Documentation/Kernel/DevelopmentGuidelines.md:211]: Oh, and this is `Set up memory ...`. "Setup" is the noun, "to set up" is the verb.

Comment (supercomputer7) [reply Documentation/Kernel/DevelopmentGuidelines.md:211]: This was taken from the actual code, but I guess it's OK to change this here :shrug:

**Pull request merged**